tests/core/index_management/test_get_settings.py

```diff
@@ -208,4 +206,176 @@ def test_custom_settings(self):
+                 "dimensions": 512,
+                 "model_location": {
+                     "s3": {
+                         "Bucket": "marqtune-public-bucket",
+                         "Key": "marqo-test-open-clip-model/epoch_1.pt",
+                     },
+                     "auth_required": False
+                 },
+                 "type": "open_clip",
+            },
+            normalizeEmbeddings=False,
+            textPreprocessing=TextPreProcessing(splitLength=3, splitMethod=TextSplitMethod.Word, splitOverlap=1)
+        ).to_marqo_index_request('a' + str(uuid.uuid4()).replace('-', ''))
+
+        indexes = self.create_indexes([
+            unstructured_marqtune_index
+        ])
+
+        marqtune_index = indexes[0]
+        expected_unstructured_custom_settings = \
+            {
+                'annParameters': {
+                    'parameters': {'efConstruction': 512, 'm': 16},
+                    'spaceType': DistanceMetric.PrenormalizedAngular
+                },
+                'filterStringMaxLength': 50,
+                'imagePreprocessing': {},
+                'model': 'marqtune/model-id/release-checkpoint',
+                'modelProperties': {
+                    "isMarqtuneModel": True,
+                },
+                'normalizeEmbeddings': False,
+                'textPreprocessing': {'splitLength': 3,
+                                      'splitMethod': TextSplitMethod.Word,
+                                      'splitOverlap': 1},
+                'audioPreprocessing': {'splitLength': 10, 'splitOverlap': 3},
+                'videoPreprocessing': {'splitLength': 20, 'splitOverlap': 3},
+                'treatUrlsAndPointersAsImages': False,
+                'treatUrlsAndPointersAsMedia': False,
+                'type': IndexType.Unstructured,
+                'vectorNumericType': VectorNumericType.Float
+            }
+
+        retrieved_index = self.config.index_management.get_index(marqtune_index.name)
+        retrieved_settings = IndexSettings.from_marqo_index(retrieved_index).dict(exclude_none=True, by_alias=True)
+        self.assertEqual(retrieved_settings, expected_unstructured_custom_settings)
+
+    def test_index_settings_should_hide_hf_model_internal_data_if_model_is_marqtune(self):
+        """Model name, dimensions, and model location should be hidden if model is marqtune
+        """
+        unstructured_marqtune_index = IndexSettings(
+            type=IndexType.Unstructured,
+            model='marqtune/model-id/release-checkpoint',
+            modelProperties= {
+                 "isMarqtuneModel": True,
+                 "dimensions": 384,
+                 "model_location": {
+                     "s3": {
+                         "Bucket": "marqtune-public-bucket",
+                         "Key": "marqo-test-hf-model/epoch_1.zip",
+                     },
+                     "auth_required": False
+                 },
+                 "type": "hf",
```

Comment:
Is this the only difference compared to the last test? Consider using subtests to reduce the duplicate test code. Also, consider add test cases for structured indexes (since there's change for structured index as well)

Reply:
Created the subtests and included both structured and unstructured test cases 👍